File: Safari-611.1.10/LayoutTests/http/tests/xmlhttprequest/resources/access-control-preflight-request-header-lowercase.php
<?php
require_once '../../resources/portabilityLayer.php';

header("Access-Control-Allow-Origin: *");
header("Access-Control-Max-Age: 0");

if ($_SERVER["REQUEST_METHOD"] == "OPTIONS") {
    // Split the Access-Control-Request-header value based on the token.
    $accessControlRequestHeaderValues = explode(", ", $_SERVER["HTTP_ACCESS_CONTROL_REQUEST_HEADERS"]);
    if (in_array("x-custom-header", $accessControlRequestHeaderValues)) // Case-sensitive comparison to make sure that browser sends the value in lowercase.
        header("Access-Control-Allow-Headers: X-Custom-Header");  // Add "Access-Control-Allow-Headers: X-Custom-Header" to "OPTIONS" response.

} else if ($_SERVER["REQUEST_METHOD"] == "GET") {
    if (isset($_SERVER["HTTP_X_CUSTOM_HEADER"]))
        echo "PASS";
    else
        echo "FAIL";
}
?>

File: LayoutTests/http/tests/xmlhttprequest/resources/access-control-preflight-request-header-lowercase.py
#!/usr/bin/env python3
import os
import sys


sys.stdout.write(
    'Content-Type: text/html\r\n'
    'Access-Control-Allow-Origin: *\r\n'
    'Access-Control-Max-Age: 0\r\n'
)

if os.environ.get('REQUEST_METHOD') == 'OPTIONS':
    if 'x-custom-header' in os.environ.get('HTTP_ACCESS_CONTROL_REQUEST_HEADERS', '').split(','):
        sys.stdout.write('Access-Control-Allow-Headers: X-Custom-Header\r\n')
    sys.stdout.write('\r\n')

elif os.environ.get('REQUEST_METHOD') == 'GET':
    sys.stdout.write('\r\n')
    if os.environ.get('HTTP_X_CUSTOM_HEADER'):
        sys.stdout.write('PASS')
    else:
        sys.stdout.write('FAIL')
